`cmake ..` fail on macOS m2/m1 , macOS x86 work well :)

<!-- You have to provide the following information whenever possible. -->

**Describe what's wrong**
<!-- A clear and concise description of what works not as it is supposed to. -->

**How to reproduce**
`cmake ..` with macOS m1/m2
and work with Brew Clang 16

**Error message and/or stacktrace**
```cpp
re@res-Mac-mini build_dev_release % ninja
[0/2] Re-checking globbed directories...
[719/2162] Building CXX object src/CMakeFiles/dbms.dir/Interpreters/Streaming/PagedAsofRowRefs.cpp.o
FAILED: src/CMakeFiles/dbms.dir/Interpreters/Streaming/PagedAsofRowRefs.cpp.o 
/opt/homebrew/bin/ccache /opt/homebrew/opt/llvm@16/bin/clang++ -DAWS_SDK_VERSION_MAJOR=1 -DAWS_SDK_VERSION_MINOR=10 -DAWS_SDK_VERSION_PATCH=36 -DBOOST_ASIO_HAS_STD_INVOKE_RESULT=1 -DBOOST_ASIO_STANDALONE=1 -DCARES_STATICLIB -DCONFIGDIR=\"\" -DENABLE_MULTITARGET_CODE=1 -DHAS_RESERVED_IDENTIFIER -DHAVE_CONFIG_H -DLIBSASL_EXPORTS=1 -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DOBSOLETE_CRAM_ATTR=1 -DOBSOLETE_DIGEST_ATTR=1 -DPLUGINDIR=\"\" -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DSASLAUTHD_CONF_FILE_DEFAULT=\"\" -DSNAPPY_CODEC_AVAILABLE -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -DUNALIGNED_OK -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/proton/build_dev_release/includes/configs -I/proton/src -I/proton/build_dev_release/src -I/proton/build_dev_release/src/Core/include -I/proton/base/base/.. -I/proton/build_dev_release/base/base/.. -I/proton/contrib/cctz/include -I/proton/base/pcg-random/. -I/proton/contrib/miniselect/include -I/proton/contrib/zstd/lib -I/proton/build_dev_release/contrib/cyrus-sasl-cmake -I/proton/build_dev_release/contrib/v8-cmake/generated -I/proton/build_dev_release/contrib/v8-cmake -I/proton/contrib/v8-cmake/v8/include -I/proton/contrib/v8-cmake/v8 -I/proton/contrib/lz4/lib -isystem /proton/build_dev_release/contrib/orc/c++/include -isystem /proton/contrib/llvm-project/libcxx/include -isystem /proton/contrib/llvm-project/libcxxabi/include -isystem /proton/contrib/libunwind/include -isystem /proton/contrib/libdivide/. -isystem /proton/contrib/jemalloc-cmake/include -isystem /proton/src/NativeLog -isystem /proton/contrib/muduo -isystem /proton/contrib/boost -isystem /proton/contrib/cityhash102/include -isystem /proton/base/poco/Net/include -isystem /proton/base/poco/Foundation/include -isystem /proton/base/poco/NetSSL_OpenSSL/include -isystem /proton/base/poco/Crypto/include -isystem /proton/contrib/boringssl/include -isystem /proton/base/poco/Util/include -isystem /proton/base/poco/JSON/include -isystem /proton/base/poco/XML/include -isystem /proton/contrib/replxx/include -isystem /proton/contrib/fmtlib-cmake/../fmtlib/include -isystem /proton/contrib/magic_enum/include -isystem /proton/contrib/double-conversion -isystem /proton/contrib/dragonbox/include -isystem /proton/contrib/re2 -isystem /proton/build_dev_release/contrib/re2-cmake -isystem /proton/contrib/zlib-ng -isystem /proton/build_dev_release/contrib/zlib-ng-cmake -isystem /proton/contrib/pdqsort -isystem /proton/contrib/xz/src/liblzma/api -isystem /proton/contrib/aws/aws-cpp-sdk-core/include -isystem /proton/build_dev_release/contrib/aws-cmake/include -isystem /proton/contrib/aws/aws-cpp-sdk-s3/include -isystem /proton/contrib/aws-c-auth/include -isystem /proton/contrib/aws-c-common/include -isystem /proton/contrib/aws-c-io/include -isystem /proton/contrib/aws-crt-cpp/include -isystem /proton/contrib/aws-c-mqtt/include -isystem /proton/contrib/aws-c-sdkutils/include -isystem /proton/contrib/snappy -isystem /proton/build_dev_release/contrib/snappy-cmake -isystem /proton/contrib/msgpack-c/include -isystem /proton/contrib/fast_float/include -isystem /proton/contrib/rocksdb/include -isystem /proton/contrib/abseil-cpp -isystem /proton/contrib/librdkafka-cmake/include -isystem /proton/contrib/librdkafka/src -isystem /proton/build_dev_release/contrib/librdkafka-cmake/auxdir -isystem /proton/contrib/croaring/cpp -isystem /proton/contrib/croaring/include -isystem /proton/contrib/krb5/src/include -isystem /proton/build_dev_release/contrib/krb5-cmake/include -isystem /proton/contrib/NuRaft/include -isystem /proton/contrib/yaml-cpp/include -isystem /proton/contrib/capnproto/c++/src -isystem /proton/contrib/arrow/cpp/src -isystem /proton/contrib/arrow-cmake/cpp/src -isystem /proton/build_dev_release/contrib/arrow-cmake/cpp/src -isystem /proton/build_dev_release/contrib/arrow-cmake/../orc/c++/include -isystem /proton/contrib/orc/c++/include -isystem /proton/build_dev_release/contrib/avro-cmake/include -isystem /proton/contrib/avro/lang/c++/api -isystem /proton/contrib/openldap-cmake/darwin_aarch64/include -isystem /proton/contrib/openldap/include -isystem /proton/contrib/sparsehash-c11 -isystem /proton/contrib/protobuf/src -isystem /proton/build_dev_release/src/Server/grpc_protos -isystem /proton/contrib/grpc/include -isystem /proton/contrib/c-ares/src/lib -isystem /proton/contrib/c-ares/include -isystem /proton/contrib/c-ares-cmake/darwin -isystem /proton/contrib/s2geometry/src -isystem /proton/contrib/libpqxx/include -isystem /proton/contrib/libpq -isystem /proton/contrib/libpq/include -isystem /proton/contrib/libstemmer_c/include -isystem /proton/contrib/wordnet-blast -isystem /proton/contrib/lemmagen-c/include -isystem /proton/contrib/simdjson/include -isystem /proton/contrib/rapidjson/include -isystem /proton/contrib/consistent-hashing -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8-a+crc -fasynchronous-unwind-tables -ffile-prefix-map=/proton=. -falign-functions=32   -Wall -Wno-unused-command-line-argument  -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-shorten-64-to-32 -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -std=gnu++2b -arch arm64 -isysroot "/Applications/Xcode-beta 2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk" -mmacosx-version-min=10.15   -D OS_DARWIN -Werror -nostdinc++ -Wno-newline-eof -Wno-non-virtual-dtor -Wno-undef -Wno-unused-parameter -Wno-documentation -Wno-undefined-reinterpret-cast -Wno-shadow-field-in-constructor -MD -MT src/CMakeFiles/dbms.dir/Interpreters/Streaming/PagedAsofRowRefs.cpp.o -MF src/CMakeFiles/dbms.dir/Interpreters/Streaming/PagedAsofRowRefs.cpp.o.d -o src/CMakeFiles/dbms.dir/Interpreters/Streaming/PagedAsofRowRefs.cpp.o -c /proton/src/Interpreters/Streaming/PagedAsofRowRefs.cpp
In file included from /proton/src/Interpreters/Streaming/PagedAsofRowRefs.cpp:1:
In file included from /proton/src/Interpreters/Streaming/PagedAsofRowRefs.h:3:
In file included from /proton/src/Interpreters/Streaming/PageBasedRowRefWithRefCount.h:3:
In file included from /proton/src/Interpreters/Streaming/RefCountDataBlockPages.h:4:
In file included from /proton/src/Interpreters/Streaming/RefCountDataBlockPage.h:3:
/proton/src/Interpreters/Streaming/RefCountDataBlock.h:32:9: error: use of undeclared identifier 'assert'
        assert(refcnt != 0);
        ^
In file included from /proton/src/Interpreters/Streaming/PagedAsofRowRefs.cpp:1:
/proton/src/Interpreters/Streaming/PagedAsofRowRefs.h:54:15: error: unknown type name 'IColumn'
        const IColumn & asof_column,
              ^
/proton/src/Interpreters/Streaming/PagedAsofRowRefs.h:61:91: error: unknown type name 'IColumn'
    const RowRefDataBlock * findAsof(TypeIndex type, ASOFJoinInequality inequality, const IColumn & asof_column, size_t row_num) const;
                                                                                          ^
/proton/src/Interpreters/Streaming/PagedAsofRowRefs.cpp:67:35: error: out-of-line definition of 'insert' does not match any declaration in 'PagedAsofRowRefs<DataBlock>'
void PagedAsofRowRefs<DataBlock>::insert(
                                  ^~~~~~
/proton/src/Interpreters/Streaming/PagedAsofRowRefs.cpp:94:30: error: out-of-line definition of 'findAsof' does not match any declaration in 'PagedAsofRowRefs<DataBlock>'
PagedAsofRowRefs<DataBlock>::findAsof(TypeIndex type, ASOFJoinInequality inequality, const IColumn & asof_column, size_t row_num) const
                             ^~~~~~~~
5 errors generated.
[724/2162] Building CXX object src/CMakeFiles/dbms.dir/Interpreters/Streaming/HashJoin.cpp.o
ninja: build stopped: subcommand failed.
re@res-Mac-mini build_dev_release % 

```

**Additional context**
<!--Add any other context about the problem here. -->

```
[8/1432] Building CXX object src/CMakeFiles/dbms.dir/Interpreters/Streaming/RefCountDataBlockPage.cpp.o
FAILED: src/CMakeFiles/dbms.dir/Interpreters/Streaming/RefCountDataBlockPage.cpp.o 
/opt/homebrew/bin/ccache /opt/homebrew/opt/llvm@16/bin/clang++ -DAWS_SDK_VERSION_MAJOR=1 -DAWS_SDK_VERSION_MINOR=10 -DAWS_SDK_VERSION_PATCH=36 -DBOOST_ASIO_HAS_STD_INVOKE_RESULT=1 -DBOOST_ASIO_STANDALONE=1 -DCARES_STATICLIB -DCONFIGDIR=\"\" -DENABLE_MULTITARGET_CODE=1 -DHAS_RESERVED_IDENTIFIER -DHAVE_CONFIG_H -DLIBSASL_EXPORTS=1 -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DOBSOLETE_CRAM_ATTR=1 -DOBSOLETE_DIGEST_ATTR=1 -DPLUGINDIR=\"\" -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DSASLAUTHD_CONF_FILE_DEFAULT=\"\" -DSNAPPY_CODEC_AVAILABLE -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -DUNALIGNED_OK -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/proton/build_dev_release/includes/configs -I/proton/src -I/proton/build_dev_release/src -I/proton/build_dev_release/src/Core/include -I/proton/base/base/.. -I/proton/build_dev_release/base/base/.. -I/proton/contrib/cctz/include -I/proton/base/pcg-random/. -I/proton/contrib/miniselect/include -I/proton/contrib/zstd/lib -I/proton/build_dev_release/contrib/cyrus-sasl-cmake -I/proton/build_dev_release/contrib/v8-cmake/generated -I/proton/build_dev_release/contrib/v8-cmake -I/proton/contrib/v8-cmake/v8/include -I/proton/contrib/v8-cmake/v8 -I/proton/contrib/lz4/lib -isystem /proton/build_dev_release/contrib/orc/c++/include -isystem /proton/contrib/llvm-project/libcxx/include -isystem /proton/contrib/llvm-project/libcxxabi/include -isystem /proton/contrib/libunwind/include -isystem /proton/contrib/libdivide/. -isystem /proton/contrib/jemalloc-cmake/include -isystem /proton/src/NativeLog -isystem /proton/contrib/muduo -isystem /proton/contrib/boost -isystem /proton/contrib/cityhash102/include -isystem /proton/base/poco/Net/include -isystem /proton/base/poco/Foundation/include -isystem /proton/base/poco/NetSSL_OpenSSL/include -isystem /proton/base/poco/Crypto/include -isystem /proton/contrib/boringssl/include -isystem /proton/base/poco/Util/include -isystem /proton/base/poco/JSON/include -isystem /proton/base/poco/XML/include -isystem /proton/contrib/replxx/include -isystem /proton/contrib/fmtlib-cmake/../fmtlib/include -isystem /proton/contrib/magic_enum/include -isystem /proton/contrib/double-conversion -isystem /proton/contrib/dragonbox/include -isystem /proton/contrib/re2 -isystem /proton/build_dev_release/contrib/re2-cmake -isystem /proton/contrib/zlib-ng -isystem /proton/build_dev_release/contrib/zlib-ng-cmake -isystem /proton/contrib/pdqsort -isystem /proton/contrib/xz/src/liblzma/api -isystem /proton/contrib/aws/aws-cpp-sdk-core/include -isystem /proton/build_dev_release/contrib/aws-cmake/include -isystem /proton/contrib/aws/aws-cpp-sdk-s3/include -isystem /proton/contrib/aws-c-auth/include -isystem /proton/contrib/aws-c-common/include -isystem /proton/contrib/aws-c-io/include -isystem /proton/contrib/aws-crt-cpp/include -isystem /proton/contrib/aws-c-mqtt/include -isystem /proton/contrib/aws-c-sdkutils/include -isystem /proton/contrib/snappy -isystem /proton/build_dev_release/contrib/snappy-cmake -isystem /proton/contrib/msgpack-c/include -isystem /proton/contrib/fast_float/include -isystem /proton/contrib/rocksdb/include -isystem /proton/contrib/abseil-cpp -isystem /proton/contrib/librdkafka-cmake/include -isystem /proton/contrib/librdkafka/src -isystem /proton/build_dev_release/contrib/librdkafka-cmake/auxdir -isystem /proton/contrib/croaring/cpp -isystem /proton/contrib/croaring/include -isystem /proton/contrib/krb5/src/include -isystem /proton/build_dev_release/contrib/krb5-cmake/include -isystem /proton/contrib/NuRaft/include -isystem /proton/contrib/yaml-cpp/include -isystem /proton/contrib/capnproto/c++/src -isystem /proton/contrib/arrow/cpp/src -isystem /proton/contrib/arrow-cmake/cpp/src -isystem /proton/build_dev_release/contrib/arrow-cmake/cpp/src -isystem /proton/build_dev_release/contrib/arrow-cmake/../orc/c++/include -isystem /proton/contrib/orc/c++/include -isystem /proton/build_dev_release/contrib/avro-cmake/include -isystem /proton/contrib/avro/lang/c++/api -isystem /proton/contrib/openldap-cmake/darwin_aarch64/include -isystem /proton/contrib/openldap/include -isystem /proton/contrib/sparsehash-c11 -isystem /proton/contrib/protobuf/src -isystem /proton/build_dev_release/src/Server/grpc_protos -isystem /proton/contrib/grpc/include -isystem /proton/contrib/c-ares/src/lib -isystem /proton/contrib/c-ares/include -isystem /proton/contrib/c-ares-cmake/darwin -isystem /proton/contrib/s2geometry/src -isystem /proton/contrib/libpqxx/include -isystem /proton/contrib/libpq -isystem /proton/contrib/libpq/include -isystem /proton/contrib/libstemmer_c/include -isystem /proton/contrib/wordnet-blast -isystem /proton/contrib/lemmagen-c/include -isystem /proton/contrib/simdjson/include -isystem /proton/contrib/rapidjson/include -isystem /proton/contrib/consistent-hashing -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8-a+crc -fasynchronous-unwind-tables -ffile-prefix-map=/proton=. -falign-functions=32   -Wall -Wno-unused-command-line-argument  -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-shorten-64-to-32 -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -std=gnu++2b -arch arm64 -isysroot "/Applications/Xcode-beta 2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk" -mmacosx-version-min=10.15   -D OS_DARWIN -Werror -nostdinc++ -Wno-newline-eof -Wno-non-virtual-dtor -Wno-undef -Wno-unused-parameter -Wno-documentation -Wno-undefined-reinterpret-cast -Wno-shadow-field-in-constructor -MD -MT src/CMakeFiles/dbms.dir/Interpreters/Streaming/RefCountDataBlockPage.cpp.o -MF src/CMakeFiles/dbms.dir/Interpreters/Streaming/RefCountDataBlockPage.cpp.o.d -o src/CMakeFiles/dbms.dir/Interpreters/Streaming/RefCountDataBlockPage.cpp.o -c /proton/src/Interpreters/Streaming/RefCountDataBlockPage.cpp
In file included from /proton/src/Interpreters/Streaming/RefCountDataBlockPage.cpp:1:
In file included from /proton/src/Interpreters/Streaming/RefCountDataBlockPage.h:3:
/proton/src/Interpreters/Streaming/RefCountDataBlock.h:13:66: error: no member named 'move' in namespace 'std'
    explicit RefCountDataBlock(DataBlock && block_) : block(std::move(block_)), refcnt(static_cast<uint32_t>(block.rows())) { }
                                                            ~~~~~^
/proton/src/Interpreters/Streaming/RefCountDataBlock.h:16:73: error: no member named 'move' in namespace 'std'
    RefCountDataBlock(RefCountDataBlock && other) noexcept : block(std::move(other.block)), refcnt(other.refcnt) { }
                                                                   ~~~~~^
/proton/src/Interpreters/Streaming/RefCountDataBlock.h:20:22: error: no member named 'move' in namespace 'std'
        block = std::move(other.block);
                ~~~~~^
In file included from /proton/src/Interpreters/Streaming/RefCountDataBlockPage.cpp:1:
/proton/src/Interpreters/Streaming/RefCountDataBlockPage.h:48:10: error: no template named 'vector' in namespace 'std'
    std::vector<RefCountDataBlock<DataBlock>> page;
    ~~~~~^
/proton/src/Interpreters/Streaming/RefCountDataBlockPage.h:53:39: error: no template named 'unique_ptr' in namespace 'std'
using RefCountDataBlockPagePtr = std::unique_ptr<RefCountDataBlockPage<RefCountDataBlock>>;
                                 ~~~~~^
In file included from /proton/src/Interpreters/Streaming/RefCountDataBlockPage.cpp:2:
/proton/src/Interpreters/Streaming/RefCountDataBlockPages.h:132:16: error: use of undeclared identifier 'RefCountDataBlockPagePtr'
    std::deque<RefCountDataBlockPagePtr<DataBlock>> block_pages;
               ^
6 errors generated.
```

--- a/src/Interpreters/Streaming/PagedAsofRowRefs.h
+++ b/src/Interpreters/Streaming/PagedAsofRowRefs.h
@@ -3,6 +3,7 @@
 #include <Interpreters/Streaming/PageBasedRowRefWithRefCount.h>
 #include <Interpreters/Streaming/SortedLookupContainer.h>
 
+#include <Columns/IColumn.h>
 #include <Core/Joins.h>
 #include <Core/Types.h>
 

--- a/src/Interpreters/Streaming/RefCountDataBlock.h
+++ b/src/Interpreters/Streaming/RefCountDataBlock.h
@@ -1,5 +1,8 @@
 #pragma once
 
+#include <cassert>
+#include <utility>
+
 namespace DB::Streaming
 {
 template <typename DataBlock>

--- a/src/Interpreters/Streaming/RefCountDataBlockPage.h
+++ b/src/Interpreters/Streaming/RefCountDataBlockPage.h
@@ -2,6 +2,8 @@
 
 #include <Interpreters/Streaming/RefCountDataBlock.h>
 
+#include <memory>
+
 namespace DB::Streaming
 {
 

